### PR TITLE
feat: More robust multi-threaded speed testing with more generalized provider support

### PR DESCRIPTION
**Pull request by mitchcapper**
New Fast.com provider is not exposed to the user currently, but once options are added it can be.  This does a better job of evaluating high speed connections and follows more closely to official testers behavior.

I have imported some of my own code here, so if it looks like overkill that is likely why.  This gets me very reliable results over  many tests.   It should also scale the test to the size of the connection, like the actual tests.

No longer directly sub-classing HttpClient as I didn't see the benefit of going that route.

### CODE REVIEW

File: SpeedtestPlugin/Classes/ByteBitSize.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        public static Int64 SmallestUnitFromUnitValue(Double unit_amt, SizeUnit unit) => _SmallestUnitFromUnitValue(unit_amt, unit);
+        public static Int64 SizeFromHumanReadable(String sizeStrWithType) => _SizeFromHumanReadable(sizeStrWithType);
+        public static Int64 SizeFromHumanReadable(String sizeStr, String sizeType) => _SizeFromHumanReadable(sizeStr, sizeType);
+        public static Int64 SizeFromHumanReadable(String sizeStr, SizeUnit sizeType) => _SizeFromHumanReadable(sizeStr, sizeType);
+        public static String HumanReadable(Double len, Int32 decimalPlaces = 2) => _HumanReadable(len, decimalPlaces);
+        public enum SizeUnit { bit = 1, kbit = 2, mbit = 3, gbit = 4, tbit = 5 }
+    }
+
+}
+namespace _IgnoreIgnoreIgnore
```
Comment (XeroxDev) [inline SpeedtestPlugin/Classes/ByteBitSize.cs]: I'm not a bug friend of such namespaces. Is there any reason for this? If no, could you please change this?

**Review by XeroxDev — CHANGES_REQUESTED** [review]
Expect the one thing, seems extremely good! Nice work!

**Review by XeroxDev — APPROVED** [review]

**Comment by XeroxDev** [comment]
I'll merge it, when I'm doing a release.

**Review by XeroxDev — CHANGES_REQUESTED** [review]
I've tested your changes and I've to inform you, that those changes are breaking the plugins functionality.

1. you're using "min_ping" to display the ping. For me it's always 0.
2. Upload throws exceptions. See https://p.tswi.me/bojojepoxa.exception for full stack trace.
3. Download is still inaccurate. Plugin displays for me 42.8 MB/s but if I use the service you've implemented (fast.com / from netflix) I get 970Mb/s a.k.a ~120MB/s

**Comment by mitchcapper** [comment]
Try again if still an issue please send the log:)

**Review by XeroxDev — APPROVED** [review]

**Pull request merged**